### PR TITLE
feat!: support Maps SDK for iOS v9

### LINKED ISSUE #475: Support Maps SDK v9
Maps SDK for iOS [v9.0.0 was released in May](https://developers.google.com/maps/documentation/ios-sdk/release-notes#May_28_2024). We need a companion version of this utility library that supports v9 (minimum iOS 15.0 supported).

### PR DESCRIPTION
**Pull request by wangela**
BEGIN_COMMIT_OVERRIDE
feat!: Raise the minimum supported iOS version to 15.0 and raise dependency on Maps SDK for iOS to v9.
END_COMMIT_OVERRIDE

Raise the minimum supported iOS version to 15.0, in line with Maps SDK for iOS v9.

Fixes #475

### CODE REVIEW

**Review by jonesfloyd — APPROVED** [review]

File: Google-Maps-iOS-Utils.podspec [hunk 1]
```diff
@@ -16,7 +16,7 @@ Pod::Spec.new do |s|
   s.changelog    = "https://github.com/googlemaps/google-maps-ios-utils/blob/main/CHANGELOG.md"
   s.license      = { :type => 'Apache 2.0', :file => 'LICENSE' }
   s.authors      = "Google Inc."
-  s.platform     = :ios, '14.0'
+  s.platform     = :ios, '15.0'
```
Comment (Benteth) [inline Google-Maps-iOS-Utils.podspec:19]: Comment

File: Podfile.template [hunk 2]
```diff
@@ -1,10 +1,10 @@
 # [START maps_ios_utils_podfile_template]
 source 'https://github.com/CocoaPods/Specs.git'
-platform :ios, '14.0'
+platform :ios, '15.0'
 
 target 'YOUR_APPLICATION_TARGET_NAME_HERE' do
   use_frameworks!
-  pod 'GoogleMaps', '8.4.0'
+  pod 'GoogleMaps', '9.0.0'
```
Comment (Benteth) [inline Podfile.template:7]: Comment

File: Podfile.template [hunk 3]
```diff
@@ -1,10 +1,10 @@
 # [START maps_ios_utils_podfile_template]
 source 'https://github.com/CocoaPods/Specs.git'
-platform :ios, '14.0'
+platform :ios, '15.0'
 
 target 'YOUR_APPLICATION_TARGET_NAME_HERE' do
   use_frameworks!
-  pod 'GoogleMaps', '8.4.0'
```
Comment (Benteth) [inline Podfile.template:7]: > _Originally posted by @Benteth in https://github.com/googlemaps/google-maps-ios-utils/pull/479#discussion_r1776139510_

**Review by Benteth — COMMENTED** [review]
> _Originally posted by @Benteth in https://github.com/googlemaps/google-maps-ios-utils/pull/479#discussion_r1776139510_

**Pull request merged**